internal/provider/network/isolated_resource.go

```diff
@@ -0,0 +1,670 @@
+							Validators: []validator.String{
+								stringvalidator.OneOf("PRIVATE", "READONLY", "READWRITE"),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *networkIsolatedResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.CloudAvenue)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *client.CloudAvenue, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *networkIsolatedResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var (
+		plan     *networkIsolatedResourceModel
+		err      error
+		urn      string
+		isGroup  bool
+		vdcGroup *govcd.VdcGroup
+		vdc      *govcd.Vdc
+	)
+
+	// Read the plan
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get Org and VDC
+	org, _, err := r.client.GetOrgAndVDC(r.client.GetOrg(), r.client.GetDefaultVDC())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving Org", err.Error())
+		return
+	}
+
+	// If VDC is not defined at data source level, use the one defined at provider level
+	if plan.OwnerID.IsNull() || plan.OwnerID.IsUnknown() {
+		if r.client.DefaultVDCExist() {
+			vdc, err = org.GetVDCByName(r.client.GetDefaultVDC(), true)
+			if err != nil {
+				resp.Diagnostics.AddError("Error retrieving VDC from Provider", err.Error())
+				return
+			}
+			// Set the OwnerID with the VDC ID
+			plan.OwnerID = types.StringValue(vdc.Vdc.ID)
+		} else {
+			resp.Diagnostics.AddError("[CREATE] Missing VDC", "OwnerID (VDC) is required when not defined at provider level")
+			return
+		}
+	} else {
+		// OwnerID is not null, so we need to retrieve the URN via Owner_ID (VDC or VDC Group)
+		// First, we need to check if it is a VDC
+		vdc, err = org.GetVDCById(plan.OwnerID.ValueString(), true)
+		if err == nil {
+			plan.OwnerID = types.StringValue(vdc.Vdc.ID)
+		} else { // It is not a VDC, so it may be a VDC Group
+			urn, err = govcd.BuildUrnWithUuid("urn:vcloud:vdcGroup:", plan.OwnerID.ValueString())
+			if err == nil {
+				vdcGroup, err = org.GetVdcGroupById(urn)
+				if err != nil {
+					resp.Diagnostics.AddError("[CREATE] Error retrieving VDC Group from OwnerID", err.Error())
+					return
+				}
+
+				// OwnerId is VDCGroup ID
+				isGroup = true
+				plan.OwnerID = types.StringValue(urn)
+			} else {
+				resp.Diagnostics.AddError("[CREATE] Error retrieving VDC or VDC Group from OwnerID", err.Error())
+				return
+			}
+		}
+	}
```

Comment:
`OwnerID` will be set only if the user not set this value otherwise there will be a change with each read. And in the description of the attribute `OwnerID` the field is described as a `UUID` and your set value to `URN`

Reply:
Yes because ID for vdcgroup is urn and ID for VDC is uuid